library/src/math/attach.rs

```diff
@@ -19,33 +20,45 @@ pub struct AttachElem {
     #[required]
     pub base: Content,
 
-    /// The top attachment.
+    /// The top attachment after the base.
     pub top: Option<Content>,
 
-    /// The bottom attachment.
+    /// The bottom attachment after the base.
     pub bottom: Option<Content>,
+
+    /// The bottom attachment before the base.
+    pub prebottom: Option<Content>,
+
+    /// The top attachment before base.
+    pub pretop: Option<Content>,
 }
 
 impl LayoutMath for AttachElem {
     fn layout_math(&self, ctx: &mut MathContext) -> SourceResult<()> {
         let base = self.base();
-        let display_limits = base.is::<LimitsElem>();
+        let display_limits = base.is::<LimitsElem>()
+            && self.prebottom(ctx.styles()).is_none()
+            && self.pretop(ctx.styles()).is_none();
         let display_scripts = base.is::<ScriptsElem>();
 
         let base = ctx.layout_fragment(&base)?;
 
+        macro_rules! take_attachment {
+            ($F: ident) => {
+                self.$F(ctx.styles())
+                    .map(|elem| ctx.layout_fragment(&elem))
+                    .transpose()?
+            };
+        }
```

Comment:
Could this be done with a closure instead of a macro?

Reply:
Yup. New patch coming!